src/Directory.Build.targets

```diff
@@ -4,14 +4,8 @@
   <Choose>
     <When Condition="'$(TargetFramework)' == 'net462' OR '$(TargetFramework)' == 'netstandard2.0' OR '$(TargetFramework)' == 'net6.0'">
       <PropertyGroup>
-        <DefineConstants>$(DefineConstants);SYSTEM_TEXT_JSON</DefineConstants>
+        <DefineConstants>$(DefineConstants);MODERN_DOTNET</DefineConstants>
```

Comment:
I wonder if there is a predefined constant we could use. No, nothing?

Reply:
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives
We could use NET462_OR_GREATER || NETSTANDARD2_0_OR_GREATER || NET6_0_OR_GREATER
to replicate same behavior and use that throughout the solution instead.

Reply:
Nah, I think it's too long. I think i won't bring as many benefits, but will definitely hurt the readability.

We might be able to define our own `NET462_OR_GREATER` but then it would introduce local behavior, inconsistent with other OSS repos.